### PR TITLE
add docs on various field generator configs

### PR DESCRIPTION
**Pull request by jaypipes**
Adds documentation for using the `is_required`, `is_read_only`, `type` and `compare` configuration fields from `generator.yaml` files.

Issue #1502

Signed-off-by: Jay Pipes <jaypipes@gmail.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/content/docs/contributor-docs/code-generator-config.md [hunk 1]
```diff
@@ -694,19 +706,83 @@ We set this `requeue_on_success_seconds` value to `60` here because the values o
...
+    hooks:
+      delta_pre_compare:
+        code: compareTags(delta, a, b)
+    fields:
+      Tags:
+        compare:
+          is_ignored: true
+```
+
+Use the `nil_equals_zero_value` subfield to instruct the code generator to treat a nil value the same as the zero value of the underlying Go type.
```
Comment (A-Hilaly) [inline docs/content/docs/contributor-docs/code-generator-config.md:785]: I'm not sure if this is implemented yet

**Review by A-Hilaly — COMMENTED** [review]
Alles gut!

File: docs/content/docs/contributor-docs/code-generator-config.md [hunk 2]
```diff
@@ -694,19 +706,83 @@ We set this `requeue_on_success_seconds` value to `60` here because the values o
...
+      MinCount:
+        is_required: false
+```
 
 ### Controlling a field's Go type
 
-**TODO**
+Use the `resources[$resource].fields[$field].type` configuration option to override a field's Go type. You will typically use this configuration option for custom fields that are not inferred by `ack-generate` by looking at the AWS API model definition.
+
+An example of this is the Policies field for a Role custom resource definition in the IAM controller. The IAM controller uses some custom hook code to allow a Kubernetes user to specify one or more Policy ARNs for a Role simply by specifying `Spec.Policies`. To define this custom field as a list of string pointers, the IAM controller's [`generator.yaml`](https://github.com/aws-controllers-k8s/iam-controller/blob/b6a62ca48c1aea7ab78e62fb3ad6845335c1f1c2/generator.yaml#L164-L168) file uses the following:
```
Comment (jljaco) [inline docs/content/docs/contributor-docs/code-generator-config.md:753]: Add some language here explaining that you have chosen to "flatten" the business logic around associating Policies with a Role here by implementing a custom field, and that this field is not actually present upstream in the API.

**Review by jljaco — APPROVED** [review]
LGTM!

File: docs/content/docs/contributor-docs/code-generator-config.md (hunk 1 above)
Comment (jaypipes) [reply docs/content/docs/contributor-docs/code-generator-config.md:785]: Yeah, it's not used in any of the generator.yaml files... I suppose I can remove it from these docs.

**Comment by jaypipes** [comment]
Closed this b/c @jljaco is pushing up a PR with more.

**Pull request closed without merging**